app/(auth)/auth/callback/route.ts

```diff
@@ -84,14 +101,20 @@ async function storeProviderTokens(user: User, session: Session) {
 	if (provider_token.startsWith("ghu_")) {
 		provider = "github";
 	}
+	if (provider_token.startsWith("ya29.")) {
```

Comment:
I couldn't find any official documentation about this behavior in a quick search.

Proposal: Use path parameters to identify the auth provider

Suggested changes:
1. Modify route file structure to use dynamic provider parameter:
   - `app/(auth)/auth/callback/[provider]/route.ts`

2. Update `getAuthCallbackUrl` to generate provider-specific paths:
   - `/auth/callback/github`
   - `/auth/callback/google`

This approach aligns better with Next.js routing conventions and improves code maintainability.

Reply:
This approach using path parameters is more robust than my previous implementation that relied on token prefix checking (`provider_token.startsWith("ghu_")`).

Reply:
> This approach aligns better with Next.js routing conventions and improves code maintainability.

thank you, I'll try that in this PR @satococoa

Reply:
I improved the provider handling by efd9f98